src/field/schema.ts

```diff
@@ -153,7 +185,13 @@ You can fix the json schema or skip this error by calling createHeadlessForm(sch
  * If it doesn't, we skip the option.
  */
 function convertToOptions(nodeOptions: JsfSchema[]): Array<FieldOption> {
-  return nodeOptions
+  const hash = hashOptions(nodeOptions)
+  const cached = optionsMap.get(hash)
+  if (cached) {
```

Comment:
Have you considered any type of unit tests here? 👀 

I understand it might be trickier because of implementation details and lifecycle. So at very least we can leave a code comment saying there's no unit tests. To carefully test manually when changing the logic.